opentelemetry-instrumentation/src/opentelemetry/instrumentation/instrumentor.py

```diff
@@ -108,7 +109,7 @@ def instrument(self, **kwargs: Any):
             conflict = self._check_dependency_conflicts()
             if conflict:
                 _LOG.error(conflict)
-                return None
+                raise DependencyConflictError(conflict)
```

Comment:
Maybe param this where it returns none except if an exec_on_conflict param is passed.